scm/github_test.go

```diff
@@ -119,111 +122,205 @@ func TestCreateHook(t *testing.T) {
+	for i := 1; i <= 5; i++ {
+		issue, cleanup := createIssue(t, s, qfTestOrg, opt.Path)
+		defer cleanup()
+		wantIssueIDs = append(wantIssueIDs, issue.IssueNumber)
 	}
-	_, err = s.CreateIssue(ctx, opt)
+
+	gotIssueIDs := []int{}
+	gotIssues, err := s.GetIssues(ctx, opt)
 	if err != nil {
 		t.Fatal(err)
 	}
+	for _, issue := range gotIssues {
+		gotIssueIDs = append(gotIssueIDs, issue.IssueNumber)
+	}
+
+	less := func(a, b int) bool { return a < b }
+	if equal := cmp.Equal(wantIssueIDs, gotIssueIDs, cmpopts.SortSlices(less)); !equal {
+		t.Errorf("scm.GetIssues() mismatch wantIssueIDs: %v, gotIssueIDs: %v", wantIssueIDs, gotIssueIDs)
+	}
 }
 
-func TestGetIssues(t *testing.T) {
+func TestGetIssue(t *testing.T) {
 	qfTestOrg := scm.GetTestOrganization(t)
 	accessToken := scm.GetAccessToken(t)
-	// Replace with Repository name
-	repo := "Replace with Repository Name"
+	qfTestUser := scm.GetTestUser(t)
 
 	// Creating new Client
-	s, err := scm.NewSCMClient(zap.NewNop().Sugar(), "github", accessToken)
-	if err != nil {
-		t.Fatal(err)
-	}
+	s := scm.NewGithubV4SCMClient(zap.NewNop().Sugar(), accessToken)
 
 	ctx := context.Background()
-
 	opt := &scm.RepositoryOptions{
 		Owner: qfTestOrg,
-		Path:  repo,
+		Path:  ag.StudentRepoName(qfTestUser),
 	}
-	_, err = s.GetRepoIssues(ctx, opt)
+
+	wantIssue, cleanup := createIssue(t, s, opt.Owner, opt.Path)
+	defer cleanup()
+
+	gotIssue, err := s.GetIssue(ctx, wantIssue.IssueNumber, opt)
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	if diff := cmp.Diff(wantIssue, gotIssue); diff != "" {
+		t.Errorf("scm.TestGetIssue() mismatch (-wantIssue +gotIssue):\n%s", diff)
+	}
 }
 
-func TestGetIssue(t *testing.T) {
+// Test case for Updating existing Issue in a git Repository
+func TestUpdateIssue(t *testing.T) {
 	qfTestOrg := scm.GetTestOrganization(t)
 	accessToken := scm.GetAccessToken(t)
-	// Replace with Repository name
-	repo := "Replace with Repository Name"
-	// Replace 0 with Issue Number in Repository
-	issueNumber := 0
+	qfTestUser := scm.GetTestUser(t)
 
 	// Creating new Client
-	s, err := scm.NewSCMClient(zap.NewNop().Sugar(), "github", accessToken)
+	s := scm.NewGithubV4SCMClient(zap.NewNop().Sugar(), accessToken)
+
+	ctx := context.Background()
+
+	opt := &scm.CreateIssueOptions{
+		Organization: qfTestOrg,
+		Repository:   ag.StudentRepoName(qfTestUser),
+		Title:        "Updated Issue",
+		Body:         "Updated Issue Body",
+	}
+
+	issue, cleanup := createIssue(t, s, opt.Organization, opt.Repository)
+	defer cleanup()
+
+	gotIssue, err := s.UpdateIssue(ctx, issue.IssueNumber, opt)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	ctx := context.Background()
+	if gotIssue.Title != opt.Title || gotIssue.Body != opt.Body {
+		t.Fatalf("scm.TestUpdateIssue() want (title: %s, body: %s), got (title: %s, body: %s)", opt.Title, opt.Body, gotIssue.Title, gotIssue.Body)
+	}
+}
 
-	opt := &scm.RepositoryOptions{
-		Owner: qfTestOrg,
-		Path:  repo,
+func TestRequestReviewers(t *testing.T) {
```

Comment:
@JosteinLindhom Did you plan to update this test as well? Seem like it could also benefit from using the `ag.StudentRepoName(qfTestUser)` thing?? Not sure.

Reply:
This one is slightly more complicated. 
To run this test I think we need to:

- Have an open pull request on `QF_TEST_ORG/QF_TEST_USER-labs`, and its number.
- Implement `RemoveRequestedReviewers` to clean up for subsequent runs.